src/vs/editor/contrib/codeAction/browser/codeActionCommands.ts

```diff
@@ -320,12 +324,48 @@ export class RefactorAction extends EditorAction {
 					: nls.localize('editor.action.refactor.noneMessage', "No refactorings available"),
 			{
 				include: CodeActionKind.Refactor.contains(args.kind) ? args.kind : CodeActionKind.None,
-				onlyIncludePreferredActions: args.preferred,
+				onlyIncludePreferredActions: args.preferred
 			},
 			args.apply);
 	}
 }
 
+export class RefactorPreview extends EditorAction {
+
+	constructor() {
+		super({
+			id: refactorPreviewCommandId,
+			label: nls.localize('refactor.preview.label', "Refactor Preview..."),
+			alias: 'Refactor Preview...',
+			precondition: ContextKeyExpr.and(EditorContextKeys.writable, EditorContextKeys.hasCodeActionsProvider),
+			description: {
+				description: 'Refactor Preview...',
+				args: [{ name: 'args', schema: argsSchema }]
+			}
+		});
+	}
+
+	public run(_accessor: ServicesAccessor, editor: ICodeEditor, userArgs: any): void {
+		const args = CodeActionCommandArgs.fromUser(userArgs, {
+			kind: CodeActionKind.Refactor,
+			apply: CodeActionAutoApply.Never
+		});
+		return triggerCodeActionsForEditorSelection(editor,
+			typeof userArgs?.kind === 'string'
```

Comment:
This block looks duplicated from the normal `Refactor` action. Can you extract it to a shared function?